src/commands/commands.h

```diff
@@ -0,0 +1,34 @@
+#pragma once
+
+#include <map> 
+#include <string>
+#include <vector>
+
+typedef int commandFunction(const std::vector<std::string>);
+
+struct CommandInfo {
+    const std::string name; 
+    const std::string usage;
+    const commandFunction* command;
```

Comment:
minor suggestion: instead of using a pointer to the function, we can use `std::function`
also, the function can take a const reference to the vector instead of a copy